### ISSUE TITLE
Integration test instructions for iOS devices

### ISSUE THREAD

**Issue by thomassth** [builttoroam/device_calendar#392]
Continuing #385 

@sowens-csd please help out when you have the time

And problems I have that isn't answered in SO:
- Do we need to have xcode installed? or just the xcode cli would be fine?
- Is the instruction the same for connected device and simulated device?

Related SO answer:
https://stackoverflow.com/questions/66514569/grant-ios-permissions-to-flutter-driver-for-integration-test

**Comment by sowens-csd** [comment]
Working on this now. I didn't notice in the PR that the test files are under the example directory. I think those should be in the root. Any problem with moving them?

**Comment by thomassth** [comment]
I thought integration tests only works when we build an app for it?

If we can test directly on the library, that would definitely save us a lot of trouble. Not sure how though.

**Comment by sowens-csd** [comment]
You're right of course. Sorry about that. I was thinking of the directory structure in an app I was working on not a plugin.

**Comment by sowens-csd** [comment]
PR #393 is ready for review. The iOS integration test uses a shell script. Currently I've moved the existing Android test to `integration_test_android.dart` but I think it might be better if we had just the one `integration_test.dart` and switched Android to use a shell script as well so that the two integration tests followed the same process. Let me know what you think once you've reviewed.

**Comment by thomassth** [comment]
Can people replace the simulated iOS device with a real one? So people with a device don't have to make a virtual one. Or there might be some situation that passes in sim but fails in actual devices.

If it's not possible then we can just write that down so people knows.

**Comment by sowens-csd** [comment]
Good question, not currently but it would be pretty easy to add an option to the script to run on a real device. Just to be sure it's clear, the user doesn't need to make a virtual device, the script does that and then deletes it after. 

Do you want me to add the real device option before we merge?
